backend/service/k8s/deployment_test.go

```diff
@@ -364,3 +365,78 @@ func TestProtoForDeploymentStatus(t *testing.T) {
 		})
 	}
 }
+
+func TestProtoForDeploymentSpecWithProbes(t *testing.T) {
+	t.Parallel()
+	var terminationVar int64 = 30
+	var deploymentTestCases = []struct {
+		id         string
+		deployment *appsv1.Deployment
+	}{
+		{
+			id: "foo",
+			deployment: &appsv1.Deployment{
+				Spec: appsv1.DeploymentSpec{
+					Template: v1.PodTemplateSpec{
+						Spec: v1.PodSpec{
+							Containers: []v1.Container{
+								{
+									Resources: v1.ResourceRequirements{
+										Limits: v1.ResourceList{
+											"cpu":    resource.MustParse("500m"),
+											"memory": resource.MustParse("128Mi"),
+										},
+										Requests: v1.ResourceList{
+											"cpu":    resource.MustParse("250m"),
+											"memory": resource.MustParse("64Mi"),
+										},
+									},
+									LivenessProbe: &v1.Probe{
+										ProbeHandler: v1.ProbeHandler{
+											HTTPGet: &v1.HTTPGetAction{
+												Path: "/",
+												Port: intstr.IntOrString{
+													IntVal: 8080,
+												},
+											},
+										},
```

Comment:
This is a good start but it only handles one case, live probe http action. Lets add more tests here for for each new case we added for live probe all handlers and the same for ready probes.

Reply:
Adding all 4 cases with all handlers